src/gt4sd/configuration.py

```diff
@@ -149,7 +169,27 @@ def get_algorithm_subdirectories_with_s3(prefix: Optional[str] = None) -> Set[st
             host=host, access_key=access_key, secret_key=secret_key, secure=secure
         )
         bucket = gt4sd_configuration_instance.gt4sd_s3_bucket
-        return client.list_directories(bucket=bucket, prefix=prefix)
+
```

Comment:
what about creating a method returning the directories taking as input the parameters needed to instantiate and query the client?

```python
def get_algorithm_subdirectories_from_s3_coordinates(prefix: Optional[str] = None, host: str, access_key: str, secret_key: str, bucket_name: str, secure: bool = True) -> Set[str]:
    clibent = GT4SDS3Client(
            host=host,
            access_key=access_key,
            secret_key=secret_key,
            secure=secure,
        )
        return client.list_directories(bucket=bucket_name, prefix=prefix)
```

And then:

```python
...
dirs = get_algorithm_subdirectories_from_s3_coordinates(...)
dirs_hub = get_algorithm_subdirectories_from_s3_coordinates(...)
versions = dirs.union(dirs_hub)
return versions
```